service/worker/workerdeployment/client.go

```diff
@@ -345,6 +345,9 @@ func (d *ClientImpl) UpdateVersionMetadata(
 		Meta:  &updatepb.Meta{UpdateId: requestID, Identity: identity},
 	})
 	if err != nil {
+		if isResourceExhausted(err) {
+			return nil, serviceerror.NewInternal(fmt.Sprintf(ErrTooManyRequests, "UpdateVersionMetadata"))
```

Comment:
hmm... not sure why we are changing the error code to Internal, if user is making rapid request we should send them ResourceExhausted error code, right?